src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisService.java

```diff
@@ -58,5 +65,23 @@ public void putCaseNoteAmendment(final long bookingId, final long caseNoteId, fi
 
         restTemplate.put(url, Map.of("text", text), bookingId, caseNoteId);
     }
+
+    public Set<Long> getScheduleActivities(final String prisonId, final LocalDate date, final TimePeriod period) {
+        final var url = "/bookings/schedules/{prisonId}/activities?date={date}&period={period}";
+
+        final var responseType = new ParameterizedTypeReference<List<Map>>() {};
+        final var response = restTemplate.exchange(url, HttpMethod.GET, null, responseType, prisonId, date, period);
+        final var body = response.getBody();
+
+        if (body == null)
+            return Collections.emptySet();
+
+        return body
+                .stream()
+                .filter(entry -> entry.containsKey("bookingId"))
+                .map(entry -> Long.valueOf(entry.get("bookingId").toString()))
```

Comment:
should this be returning the booking ids? - not scheduled activity ids or something?  
(or is it a different type of booking? - like can you be booked on an activity?)

Reply:
Yeah, that's correct. Only the booking ids are required to pull the attendance information.

Reply:
would be better to indicate that in the method name e.g. `getBookingIdsForScheduleActivities`

Reply:
Ah, crap yes!